### PR TITLE
Output cluster data as a set of blocks

### PR DESCRIPTION
**Pull request by vrutkovs**
Use https://slack.dev/python-slackclient/basic_usage.html#formatting-with-block-kit to output cluster details instead of a plain file upload. This makes a replied message show links and potentially use markdown

### CODE REVIEW

File: cluster-support-bot.py [hunk 1]
```diff
@@ -142,40 +142,51 @@ def handle_help(payload, args=None, body=None, subparser=None):
...
         summary, _ = get_summary(cluster=cluster)
+        for line in summary:
+            blocks.append(_block_from_text(line))
     except ValueError as error:
         return web_client.chat_postMessage(
             channel=channel,
             thread_ts=thread,
             text='{} {}'.format(cluster, error))
-    return web_client.chat_postMessage(channel=channel, thread_ts=thread, text=summary)
-
```
Comment (wking) [inline cluster-support-bot.py:159]: You want to keep this blank line for PEP 8's [two lines between top-level functions][1].

[1]: https://www.python.org/dev/peps/pep-0008/#blank-lines

Comment (vrutkovs) [reply cluster-support-bot.py:159]: done

File: cluster-support-bot.py [hunk 2]
```diff
@@ -142,40 +142,53 @@ def handle_help(payload, args=None, body=None, subparser=None):
     message = subparser.format_help()
     return web_client.chat_postMessage(channel=channel, thread_ts=thread, text=message)
 
+def _block_from_text(line):
```
Comment (wking) [inline cluster-support-bot.py:145]: nit: also want two blank lines around this root-level function for PEP 8

**Comment by wking** [comment]
Implementation looks fine to me.  Not clear to me if we want this or not.  Probably for `summary`.  Maybe not for `detail` (in which case we'd need to rework `detail` to decouple a bit from the `summary` implementation).  My concern is that in some crazy world where folks make a bunch of comments on a cluster, `detail` might grow too large to be a Slack comment (which you cannot collapse?), while with an attachment we can grow as long as we want.  Thoughts?

**Comment by vrutkovs** [comment]
Right, I'll rework this:
* some content has defined size - clsuter ID, subscription, dashboard links - these would be output using markdown
* Summary and notes could grow large, these should be output as attach if these are larger than n lines (3?)

**Comment by wking** [comment]
> Summary and notes could grow large, these should be output as attach if these are larger than n lines (3?)

Sounds good to me.  Also, I usually just write one or two big, paragraph-scale lines for a summary body, so it might be worth setting a char cap ~300 as well on what can be inlined.

**Comment by vrutkovs** [comment]
Ah, Slack won't let me upload the summary as a file and display it there, cause it only supports remote files in the API

**Comment by vrutkovs** [comment]
Updated, Slack would automatically add "Show more" link if the message is too large due to the summary

Screenshot-20200411112442-465x548

File: cluster-support-bot.py [hunk 3]
```diff
@@ -143,39 +142,67 @@ def handle_help(payload, args=None, body=None, subparser=None):
...
             channel=channel,
             thread_ts=thread,
             text='{} {}'.format(cluster, error))
-    entries = [summary]
-    for note in notes:
-        entries.append(
-            '{subject}\n{body}'.format(**note),
-        )
-    return web_client.files_upload(channels=channel, thread_ts=thread, content='\n\n'.join(entries))
+    web_client.chat_postMessage(channel=channel, thread_ts=thread, blocks=blocks)
```
Comment (wking) [inline cluster-support-bot.py]: can we keep the `return`?

Comment (vrutkovs) [reply cluster-support-bot.py]: Fixed

**Pull request merged**